controllers/imagecollector/imagecollector_controller.go

```diff
@@ -0,0 +1,342 @@
+			return res, err
+		}
+		return ctrl.Result{RequeueAfter: repeatPeriod}, nil
+	}
+
+	// else length is 1, so check job phase
+	switch phase := relevantJobs[0].Status.Phase; phase {
+	case eraserv1alpha1.PhaseCompleted:
+		log.Info("completed phase")
+		if relevantJobs[0].Status.DeleteAfter == nil {
+			if res, err := r.updateSharedCRD(ctx, req, imageCollector); err != nil {
+				return res, err
+			}
+			relevantJobs[0].Status.DeleteAfter = util.After(time.Now(), *util.SuccessDelDelaySeconds)
+			if err := r.Status().Update(ctx, &relevantJobs[0]); err != nil {
+				log.Info("Could not update Delete After for job " + relevantJobs[0].Name)
+			}
+		}
+		if res, err := r.handleJobDeletion(ctx, &relevantJobs[0]); err != nil || res.RequeueAfter > 0 {
+			return res, err
+		}
+	case eraserv1alpha1.PhaseFailed:
+		log.Info("failed phase")
+		if relevantJobs[0].Status.DeleteAfter == nil {
+			relevantJobs[0].Status.DeleteAfter = util.After(time.Now(), *util.ErrDelDelaySeconds)
+			if err := r.Status().Update(ctx, &relevantJobs[0]); err != nil {
+				log.Info("Could not update Delete After for job " + relevantJobs[0].Name)
+			}
+		}
+		if res, err := r.handleJobDeletion(ctx, &relevantJobs[0]); err != nil || res.RequeueAfter > 0 {
+			return res, err
+		}
+	default:
+		log.Error(errors.New("should not reach this point for imagejob"), "imagejob: ", relevantJobs[0])
+	}
+
+	log.Info("done reconcile")
+
+	return ctrl.Result{RequeueAfter: repeatPeriod}, nil
+}
+
+func isNotFound(err error) bool {
+	if err != nil && client.IgnoreNotFound(err) == nil {
+		return true
+	}
+	return false
+}
+
+func (r *Reconciler) handleJobDeletion(ctx context.Context, job *eraserv1alpha1.ImageJob) (ctrl.Result, error) {
+	log.Info("start job deletion")
+	if job.Status.DeleteAfter == nil {
+		log.Info("delete after is nil")
+		return ctrl.Result{}, nil
+	}
+
+	until := time.Until(job.Status.DeleteAfter.Time)
+	if until > 0 {
+		log.Info("Delaying imagejob delete", "job", job.Name, "deleteAter", job.Status.DeleteAfter)
+		return ctrl.Result{RequeueAfter: until}, nil
+	}
+
+	log.Info("Deleting imagejob", "job", job.Name)
+	err := r.Delete(ctx, job)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	log.Info("end job deletion")
+	return ctrl.Result{}, nil
+}
+
+func (r *Reconciler) createImageJob(ctx context.Context, req ctrl.Request, imageCollector *eraserv1alpha1.ImageCollector) (ctrl.Result, error) {
+	job := &eraserv1alpha1.ImageJob{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "imagejob-",
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(imageCollector, schema.GroupVersionKind{Group: "eraser.sh", Version: "v1alpha1", Kind: "ImageCollector"}),
+			},
+		},
+		Spec: eraserv1alpha1.ImageJobSpec{
+			JobTemplate: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					RestartPolicy: corev1.RestartPolicyNever,
+					Containers: []corev1.Container{
+						{
+							Name:            "collector",
+							Image:           *collectorImage,
```

Comment:
where is this being set? we need something like https://github.com/Azure/eraser/blob/main/config/manager/kustomization.template.yaml#L18-L24